### ISSUE TITLE
502.5 errors (error code 0x800705b4) after app runs for a while

### ISSUE THREAD

**Issue by wags** [aspnet/IISIntegration#369]
Hi,
I have been chasing this issue for several days now. The scenario seems to align with [this section](http://jakeydocs.readthedocs.io/en/latest/publishing/iis.html#iis-integration-middleware-not-implemented-or-useurls-after-useiisintegration) of the documentation, but my attempts at fixing it have been unsuccessful so far.

I have a fairly basic web application, let's call it Rover, deployed to IIS on a Windows 2012 server. The app starts properly and runs for a while. After some time (hours), the app can become unresponsive and browsers report a 502.5 error. I have migrated from .NET Core 1.0 to 1.1.1, and recently upgraded to 1.1.2. None of these updates have resolved the issue. I have basic Serilog and stdout logging configured, and nothing has jumped out at me in those logs.

The Windows Event Viewer provides the most insight. The series of events usually starts with a warning from **IIS AspNetCore Module**. It reads something like `Failed to gracefully shutdown process '2264'.`. (The process number varies.) After that, a couple dozen errors appear, again from the same source as the warning, about two minutes apart. These all read:
```
Application 'MACHINE/WEBROOT/APPHOST/ROVER' with physical root 'C:\inetpub\wwwroot\rover\'
created process with commandline '"C:\Program Files\dotnet\dotnet.exe" .\Rover.dll' but either
crashed or did not reponse or did not listen on the given port '10343', ErrorCode = '0x800705b4'
```
Again, with various port numbers.

I can't say for sure that it isn't an issue in my app, but it really doesn't have any fancy or custom configuration.

My `Program.cs` file uses the default Main method:
```
public static void Main(string[] args)
{
    var host = new WebHostBuilder()
        .UseKestrel()
        .UseContentRoot(Directory.GetCurrentDirectory())
        .UseIISIntegration()
        .UseStartup<Startup>()
        .Build();

    host.Run();
}
```

As I've scoured around for answers, I have seen a couple other reports of this error but no solution. Some of the other issues here also seem similar, but don't reference error code `0x800705b4` specifically.

I have also tried messing with web.config a bit. I hard-coded the full path to `dotnet.exe` (i.e. `C:\Program Files\dotnet\dotnet.exe`) in `processPath`.

Is there more I can do to help determine the root cause of this bug? I'd be happy to set up tests, share more code, etc. Just point me in the right direction!

I'd also like to ping @Tratcher directly since @PinpointTownes sent me your way 😄 

Thanks!

**Comment by Tratcher** [comment]
@pan-wang for debugging steps.

**Comment by wags** [comment]
Hi @Tratcher, do you know anyone else that might be able to point me in the right direction?

**Comment by pan-wang** [comment]
error code 0x800705b4 means "This operation returned because the timeout period expired". it indicates the backend did not response on time (default timeout value is 2 minutes). You can confirm this timeout by enabling IIS FREB log (only log 502.x error).  I can take a look at the FREB log if you need help.
You application seems to have scalability issue. You may want to turn on Kestrel log to see what happens and capture a dump for analysis.

**Comment by wags** [comment]
@pan-wang thank you for the tip! I will set up that IIS FREB log and try to pinpoint exactly where it is failing. Part of the app retrieves data from an API and I have noticed some occasional extreme latency on its responses. There might be a pattern there.

**Comment by wags** [comment]
After digging into this one further, and with some luck, I believe the culprit was the antivirus solution on the server. The web server is a VM and the antivirus solution is running at the VMware host level.

**Comment by wags** [comment]
It is safe to close this issue. The problem was definitely on my end, and it was due to an outdated version of antivirus on the server that was incompatible with my VMware environment. I have not seen the problem since that got updated. Thanks!

**Comment by PinpointTownes** [comment]
@wags glad you finally found the root cause of your issue!

Curious: what's the antivirus you're using? (you can PM me if you prefer)

**Comment by wags** [comment]
@PinpointTownes Thanks!

We are using antivirus from Sophos. We were originally using their _Sophos Antivirus for VMware vShield_ product. As newer versions of VMware were released (6.0+), Sophos rolled out an entirely new solution called `Sophos for Virtual Environments`. In fact, the previous solution is facing end of life fairly soon, if not already.

After we did the upgrade to Sophos for Virtual Environments, we never saw the issue again.
